### PR TITLE
support ^^ at CTFE using std.math.pow as intrinsic

### PR DESCRIPTION
**Pull request by UplinkCoder**
As per title.
This fixes Pow symptomatically.

### CODE REVIEW

File: src/root/ctfloat.d [hunk 1]
```diff
@@ -59,6 +59,16 @@ extern (C++) struct CTFloat
             assert(0);
     }
 
+    //only define pow if we can import phobos
+    static if (__traits(compiles, () {import std.math : pow; return pow(1.0,1.0);}))
```
Comment (ibuclaw) [inline src/root/ctfloat.d]: When would this ever not be true?

**Comment by UplinkCoder** [comment]
@klickverbot That does not work.
Id._pow is a special identifier reserved for the rewritten call.
It is not used as a regular builtin.

**Comment by UplinkCoder** [comment]
@klickverbot yeah it's a pretty bad hack.
I was surprised to find that in dmd.

**Comment by UplinkCoder** [comment]
Builtins are resolved after the call is interpreted, which means that dmd will die because of the disallowed cast before it has time to check for the builtin.

**Comment by UplinkCoder** [comment]
... Then I have to fix it properly :)

**Pull request closed without merging**